pkg/queue/manager.go

```diff
@@ -103,6 +103,13 @@ func (m *Manager) DeleteQueue(q *kueue.Queue) {
 	delete(m.queues, key)
 }
 
+func (m *Manager) Status(q *kueue.Queue) (int, error) {
```

Comment:
`m.RLock()`

Reply:
since queue already has a defined mutex, and is needed for the cond to broadcast, I am going with it, let me know if that is not enough

Reply:
yeah, we should definitely use the same lock, but can you swap it for a RWLock?

Reply:
Did you check if this was possible?